MotionPhotoReader/app/src/androidTest/java/com/google/android/libraries/motionphotoreader/MotionPhotoReaderTest.java

```diff
@@ -0,0 +1,179 @@
+package com.google.android.libraries.motionphotoreader;
+
+import android.graphics.Bitmap;
+import android.media.MediaExtractor;
+import android.os.Bundle;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.adobe.internal.xmp.XMPException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.stubbing.Answer;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Instrumented test for MotionPhotoReader class.
+ */
+@RunWith(AndroidJUnit4.class)
+public class MotionPhotoReaderTest {
+    private static final String filename = "/sdcard/MVIMG_20200621_200240.jpg";
+    private static final int NUM_FRAMES = 44;
+    private static final long SEEK_AMOUNT_US = 10_000L;
+
+    // TODO: close motion photo readers
+    @Before
+    public void setUp() {
+
+    }
```

Comment:
Remove the Before and setup block since not needed. And then let's do this to enforce closing the motion readers:

```
private final List<Runnable> cleanup = new ArrayList<>();

@After
public void tearDown() {
  for (Runnable r : cleanup) {
    r.run();
  }
  cleanup.clear();
}
```


And then in each test, we can add the cleanup needed like this:
```
MotionPhotoReader reader = MotionPhotoReader.open( ...);
cleanup.add(reader::close);
```